lib/msal-browser/src/app/PublicClientApplication.ts

```diff
@@ -417,3 +415,40 @@ export class PublicClientApplication implements IPublicClientApplication {
         return this.controller.clearCache(logoutRequest);
     }
 }
+
+/**
+ * creates NestedAppAuthController and passes it to the PublicClientApplication,
+ * falls back to StandardController if NestedAppAuthController is not available
+ *
+ * @param configuration
+ * @returns IPublicClientApplication
+ *
+ */
+export async function createNestablePublicClientApplication(
+    configuration: Configuration
+): Promise<IPublicClientApplication> {
+    const nestedAppAuth = new NestedAppOperatingContext(configuration);
+    await nestedAppAuth.initialize();
+
+    if (nestedAppAuth.isAvailable()) {
+        const controller = new NestedAppAuthController(nestedAppAuth);
+        return new PublicClientApplication(configuration, controller);
+    }
+
+    return createStandardPublicClientApplication(configuration);
+}
+
+/**
+ * creates PublicClientApplication using StandardController
+ *
+ * @param configuration
+ * @returns IPublicClientApplication
+ *
+ */
+export async function createStandardPublicClientApplication(
```

Comment:
why not reuse `createPublicClientApplication`? seems like `createPublicClientApplication` and `createStandardPublicClientApplication` do basically the same thing and end up using StandardController?

Reply:
Forward looking. We're thinking long term the plan should be to use these create___ functions and the static method on PCA will be removed. That allows us to provide a way to treeshake out any and all controllers that aren't being used by a given app, which isn't possible today with the constructor & static create method referencing the Standard Controller.

Reply:
I will follow up with cleaning controller factory and other code that is not relevant soon.